### PR TITLE
Check BuildType for RelWDbgInfo

### PR DESCRIPTION
**Pull request by oberbichler**

### CODE REVIEW

File: applications/IgaApplication/tests/node_surface_geometry_3d_tests.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def testValueAt(self):
             self.assertAlmostEqual(displacement_z_at_t, t - t**2)
 
     def testNodeIndexOutOfRange(self):
-        if Kernel.BuildType() == 'Release':
+        if Kernel.BuildType() in ['Release', 'RelWDbgInfo']:
```
Comment (philbucher) [inline applications/IgaApplication/tests/node_surface_geometry_3d_tests.py]: I suggest to use:
If „Debug“ in Kernel.BuildType():

**Comment by oberbichler** [comment]
Writing down the effected build types explicitly makes it easier to understand what is happening.

Btw. In your mail you write `RelWDbgInfo` but in the cmake docs I can only find `RelWithDebInfo` as a default build type. Is `RelWDbgInfo` a custom build type inside Kratos?

**Comment by philbucher** [comment]
I didn’t remember the exact name, sorry

If you want to define the full name then I would check against Debug and FullDebug
Otherwise it can fail when there is a custom build type
Afaik currently used in the nightly tests right @roigcarlo ?

**Comment by jcotela** [comment]
This is the definition of build types in kratos:
https://github.com/KratosMultiphysics/Kratos/blob/dbdcea0e5ebb704bb06d4d6b86671a5c0006e721/CMakeLists.txt#L24

**Comment by jcotela** [comment]
@philbucher yes, the Custom mode is used to set special things in automated builds (I think also in Travis/appveyor)

**Comment by philbucher** [comment]
Then I guess Debug or FullDebug is the safe option
If not you can talk to each other :)

**Comment by roigcarlo** [comment]
As @jcotela  said custom is for special things. Specifically:
Travis -> Custom
Appveyor -> Debug
Nightly -> Custom

In both Travis and Nightly custom means a default Release build without `-g` or `-pg` but enforcing `-O0` (by default release uses `-O2` or `-O3`) so the compilation time is reduced considerable.

We only use Debug in win because visual studio is special, unique and wonderfull. 
The reason to use Release + O0 is that I prefer to test the code that is closer to what the final user will have.

If it is interesting for you (specially when I finish to setup the jenkings machine) we can set up a matrix of build that tests things in release and debug and maybe different compilers, but right now we still limited by the times.

**Comment by philbucher** [comment]
thanks @roigcarlo 
@oberbichler now you can decide what you prefer ;)

**Comment by RiccardoRossi** [comment]
minor comment: can u use the same names we use for the cmake?

i mean
Release, RelWithDebInfo, FullDebug, Debug

etc?

**Comment by oberbichler** [comment]
@philbucher 
> Otherwise it can fail when there is a custom build type

Custom build types are unpredictable in this context anyway. Even when we check for `Debug` (Imagine `CustomBuildTypeWithNDebug` or the user changes the flags for a build type...). Let's keep it simple and check only the build types we are using.

Usually I try to use `If condition=true Then Exit` to check for the 'special case'. At least my compiler is checking assertion by default so I would say the case were assertions are disabled is the 'special case'.

Anyway: too much discussion for one line of code. As long as it works for `Release` and `Debug` I am fine. So feel free to change it however you want.

**Comment by oberbichler** [comment]
@roigcarlo jenkins is comming?

**Review by jcotela — APPROVED** [review]
Approving.
@RiccardoRossi the implementation in this PR should be compatible with the existing Kratos build mode names.

**Pull request merged**